### PR TITLE
BC-4455 - Injection error for "envConfigModule"

### PR DESCRIPTION
**Pull request by bergatco**
# Short Description
QA note: 
Change tested, no issues found, for more details see jira ticket: https://ticketsystem.dbildungscloud.de/browse/BC-4455

@christian-darsow pointed out to us, that by https://github.com/hpi-schul-cloud/nuxt-client/pull/2607/commits/74e1a1e0a01e2eb5eba8d4d79a172a446fac976a we introduced an injection error within the "LanguageMenu".

## Links to Ticket and related Pull-Requests

https://ticketsystem.dbildungscloud.de/browse/BC-4455

Link to deployment : https://default-bc-4455-injection-error-for-envconfigmodule.cd.dbildungscloud.dev/

@SevenWaysDP original changes : https://github.com/hpi-schul-cloud/nuxt-client/pull/2607/commits/74e1a1e0a01e2eb5eba8d4d79a172a446fac976a

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -133,6 +133,7 @@ import {
 			contentModule,
 			contextExternalToolsModule,
 			copyModule,
+			envConfigModule,
```
Comment (uidp) [inline src/main.ts:136]: I don't think we should inject the module twice. We should rather refactor the "old" OptionsAPI injects to using the injection keys, e.g.:
```typescript
inject: {
	notifierModule: { from: NOTIFIER_MODULE_KEY },
	i18n: { from: I18N_KEY },
},
```

**Review by uidp — CHANGES_REQUESTED** [review]

File: src/components/molecules/LanguageMenu.unit.ts [hunk 2]
```diff
@@ -17,7 +18,7 @@ describe("@/components/templates/LanguageMenu", () => {
 			}),
 			provide: {
 				authModule: authModuleMock,
-				envConfigModule: envConfigModuleMock,
+				[ENV_CONFIG_MODULE_KEY as symbol]: envConfigModuleMock,
```
Comment (uidp) [inline src/components/molecules/LanguageMenu.unit.ts:21]: `ENV_CONFIG_MODULE_KEY.valueOf()`?

File: src/pages/user-login-migration/UserLoginMigrationError.page.unit.ts [hunk 3]
```diff
@@ -44,7 +45,7 @@ describe("UserLoginMigrationError", () => {
 				}),
 				provide: {
 					systemsModule,
-					envConfigModule,
+					[ENV_CONFIG_MODULE_KEY as symbol]: envConfigModule,
```
Comment (uidp) [inline src/pages/user-login-migration/UserLoginMigrationError.page.unit.ts:48]: `ENV_CONFIG_MODULE_KEY.valueOf()`?

**Review by uidp — APPROVED** [review]

File: src/components/molecules/LanguageMenu.unit.ts (hunk 2 above)
Comment (bergatco) [reply src/components/molecules/LanguageMenu.unit.ts]: Fixed!

File: src/pages/user-login-migration/UserLoginMigrationError.page.unit.ts (hunk 3 above)
Comment (bergatco) [reply src/pages/user-login-migration/UserLoginMigrationError.page.unit.ts]: Fixed!

**Review by uidp — APPROVED** [review]

**Pull request merged**